mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareSchedulerActor.java

```diff
@@ -382,4 +410,25 @@ private static class Noop {
     private String getSchedulingQueueKeyFor(WorkerId workerId) {
         return "Retry-Schedule-Request-For" + workerId.toString();
     }
+
+    private Set<String> getJobClustersWithArtifactCacheEnabled() {
+        String jobClusters =
+            ServiceRegistry.INSTANCE.getPropertiesService()
+                .getStringValue("mantis.artifactCaching.jobClusters", "");
+        return new HashSet<>(Arrays.asList(jobClusters.split(",")));
+    }
+
```

Comment:
The semantics of this feels very specific to this implementation. I wonder if there's a way for us to make this more extendable.